### ISSUE TITLE
My game crashed

### ISSUE THREAD

**Issue by danielrae83** [OpenRA/OpenRA#18145]
## System Information

- Operating System:  Windows 10
- OpenRA Version: 20200503
- OpenRA Mod: Red alert
- Source: Official download package

## Exception log

```
OpenRA engine version release-20200503
Red Alert mod version release-20200503
on map db2e8110eb509224e7a84a8ae6b1446e82c35a29 (Giant Desert lake with creeps by Daniel Rae).
Date: 2020-05-18 16:10:23Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.IndexOutOfRangeException`: Index was outside the bounds of the array.
   at OpenRA.Mods.Common.Traits.Locomotor.GetCache(CPos cell)
   at OpenRA.Mods.Common.Traits.Locomotor.CanStayInCell(CPos cell)
   at OpenRA.Mods.Common.Traits.MobileInfo.CanStayInCell(World world, CPos cell)
   at OpenRA.Mods.Common.Activities.MoveAdjacentTo.<CandidateMovementCells>b__14_0(CPos c)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at OpenRA.Mods.Common.Activities.MoveAdjacentTo.CalculatePathToTarget(Actor self, BlockedByActor check)
   at OpenRA.Mods.Common.Activities.Move.EvalPath(BlockedByActor check)
   at OpenRA.Mods.Common.Activities.Move.OnFirstRun(Actor self)
   at OpenRA.Activities.Activity.TickOuter(Actor self)
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act)
   at OpenRA.Activities.Activity.TickChild(Actor self)
   at OpenRA.Mods.Common.Activities.MoveAdjacentTo.Tick(Actor self)
   at OpenRA.Activities.Activity.TickOuter(Actor self)
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act)
   at OpenRA.Activities.Activity.TickChild(Actor self)
   at OpenRA.Activities.Activity.TickOuter(Actor self)
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act)
   at OpenRA.Actor.Tick()
   at OpenRA.World.Tick()
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)


```

## Additional information
Hi guys, long time fan of OpenRA, first time posting to this. I create custom maps regularly, usually really large ones designed for 20-30 players (I often go with 8 teams of 3) on very large, resource-rich maps.

Anyway, created a modified version of an older desert based map with lots of "creep" bases and I get a crash within the first few mins.

Any ideas what this exception log means and what I can do to fix it? Guessing it's to do with a CPU/Creep actor.

Thanks and keep up the amazing work.

Daniel
[debug.log](https://github.com/OpenRA/OpenRA/files/4645341/debug.log)
[exception-2020-05-18T161023Z.log](https://github.com/OpenRA/OpenRA/files/4645343/exception-2020-05-18T161023Z.log)

**Comment by abcdefg30** [comment]
Hi and thanks for the report. :) This issue was resolved by #18083 and will be fixed in our next release.

**Comment by danielrae83** [comment]
Thank you!
